site/public/js/docker_interface.js

```diff
@@ -17,3 +17,95 @@ function collapseSection(id,btn_id) {
         btn.innerHTML = 'Collapse';
     }
 }
+
+function filterOnClick() {
+    const this_filter = new Map();
+
+    if ($(this).hasClass('fully-transparent')) {
+        $(this).removeClass('fully-transparent');
+    }
+    else {
+        $(this).addClass('fully-transparent');
+    }
+
+    $(".filter-buttons").each(function (){
+        this_filter.set($(this).data("capability"), !$(this).hasClass("fully-transparent"));
+    });
+
+    $(".image-row").each(function() {
+        const this_row = $(this);
+        let hide = true;
+        if ($(this).find(".badge").length == 0) {
+            hide = false;
+        }
+        $(this).find(".badge").each(function (){
+            if (this_filter.get($(this).text())) {
+                hide = false;
+            }
+        });
+        if (hide) {
+            this_row.hide();
+        }
+        else {
+            this_row.show();
+        }
+    });
+}
+
+function addFieldOnChange() {
+    const command = $(this).val();
+    const regex = new RegExp('^[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+/[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+:[a-zA-Z0-9][a-zA-Z0-9._-]{0,127}$');
+    if (!regex.test(command)) {
+        $('#send-button').attr('disabled',true);
+        $('#docker-warning').css('display', '');
+    }
+    else {
+        $('#send-button').attr('disabled',false);
+        $('#docker-warning').css('display', 'none');
+    }
+}
+
+function addImage(url) {
+    const capability = $('#capability-form').val();
+    const image = $('#add-field').val();
+    $.ajax({
+        url: url,
+        type: 'POST',
+        data: {
+            'capability': capability,
+            'image': image,
+            csrf_token: csrfToken
+        },
+        success: function(data) {
+            console.log(data);
+            location.reload();
+        },
+        error: function(err) {
```

Comment:
This error field will only ever get called if the ajax call throws a javascript error and not if the server responds with a 500 error or does not succeed. 

If given a 500 error the json response will become a frog robot and cannot be parsed, otherwise you'll get a standard response : https://submitty.org/developer/json_responses